### PR TITLE
2811 update agent repository get plugin

### PR DESCRIPTION
**Pull request by cakekoa**
# What does this PR do?

Fixes part of #2811. Merge after #2835

Updates the plugin repository to get plugins filtered by operating system

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~~Was the CHANGELOG.md updated to reflect the changes?~~
* [ ] ~~Was the documentation framework updated to reflect the changes?~~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the unit tests
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/repositories/agent_plugin_repository_caching_decorator.py [hunk 1]
```diff
@@ -18,6 +19,14 @@ def __init__(self, agent_plugin_repository: IAgentPluginRepository):
...
         return self._agent_plugin_repository.get_plugin(plugin_type, name)
 
+    @lru_cache()
+    def get_plugin_for_os(
+        self, host_operating_system: OperatingSystem, plugin_type: AgentPluginType, name: str
+    ) -> AgentPlugin:
+        return self._agent_plugin_repository.get_plugin_for_os(
+            host_operating_system, plugin_type, name
+        )
+
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repositories/agent_plugin_repository_caching_decorator.py:29]: Why do we still have `get_plugin()`? When will we use it?

File: monkey/monkey_island/cc/repositories/i_agent_plugin_repository.py [hunk 2]
```diff
@@ -20,9 +23,19 @@ def get_plugin(self, plugin_type: AgentPluginType, name: str) -> AgentPlugin:
         pass
 
     @abstractmethod
-    def get_plugin_catalog(self) -> Sequence[Tuple[AgentPluginType, str]]:
+    def get_all_plugin_config_schemas(self) -> Dict[AgentPluginType, Dict[str, Dict[str, Any]]]:
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/repositories/i_agent_plugin_repository.py:26]: For consistency

```suggestion
    def get_all_plugin_configuration_schemas(self) -> Dict[AgentPluginType, Dict[str, Dict[str, Any]]]:
```

File: monkey/tests/monkey_island/in_memory_agent_plugin_repository.py [hunk 3]
```diff
@@ -1,20 +1,45 @@
...
+
+        return schemas
+
+    def get_all_plugin_manifests(self) -> Dict[AgentPluginType, Dict[str, AgentPluginManifest]]:
+        manifests: Dict[AgentPluginType, Dict[str, AgentPluginManifest]] = {}
+
+        for plugin in self._plugins.values():
+            plugin_type = plugin.plugin_manifest.plugin_type
+            if plugin_type not in manifests.keys():
+                manifests[plugin_type] = {}
```
Comment (mssalvatore) [inline monkey/tests/monkey_island/in_memory_agent_plugin_repository.py:39]: Use `setdefault()`

File: monkey/tests/monkey_island/in_memory_agent_plugin_repository.py [hunk 4]
```diff
@@ -1,20 +1,45 @@
...
 
-    def get_plugin_catalog(self) -> Sequence[Tuple[AgentPluginType, str]]:
-        return [(AgentPluginType.EXPLOITER, plugin_name) for plugin_name in self._plugins.keys()]
+    def get_all_plugin_config_schemas(self) -> Dict[AgentPluginType, Dict[str, Dict[str, Any]]]:
+        schemas: Dict[AgentPluginType, Dict[str, Dict[str, Any]]] = {}
+
+        for plugin in self._plugins.values():
+            plugin_type = plugin.plugin_manifest.plugin_type
+            if plugin_type not in schemas.keys():
+                schemas[plugin_type] = {}
```
Comment (mssalvatore) [inline monkey/tests/monkey_island/in_memory_agent_plugin_repository.py:28]: Use `setdefault()`

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Replace all "config" with "configuration

File: monkey/monkey_island/cc/resources/agent_plugins_manifest.py [hunk 5]
```diff
@@ -32,11 +32,10 @@ def get(self, plugin_type: str, name: str):
...
 
         try:
-            agent_plugin_manifest = self._agent_plugin_repository.get_plugin(
-                plugin_type=agent_plugin_type, name=name
-            ).plugin_manifest
+            plugin_manifests = self._agent_plugin_repository.get_all_plugin_manifests()
+            agent_plugin_manifest = plugin_manifests[agent_plugin_type][name]
             return make_response(agent_plugin_manifest.dict(simplify=True), HTTPStatus.OK)
-        except UnknownRecordError:
+        except (KeyError):
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/agent_plugins_manifest.py:38]:
```suggestion
        except KeyError:
```

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**